src/mesh/rbf.f90

```diff
@@ -0,0 +1,597 @@
+! Build mapping between 1D and tensor product ordering for a bi-linear surface
+CALL getBasisMappingHexa(1,nAns,HexaMapLinear,HexaMapInvLinear)
+DEALLOCATE(HexaMapLinear)
+
+
+!--------------------------------------------------------------------------------------------------------!
+! Count number of control points, needed to allocate space etc.
+!--------------------------------------------------------------------------------------------------------!
+
+! Reset the tmp variable later used to mark already visited control points
+Elem=>FirstElem
+DO WHILE(ASSOCIATED(Elem))
+  Side=>Elem%firstSide
+  DO WHILE(ASSOCIATED(Side))
+    IF(.NOT.ASSOCIATED(Side%BC)) THEN
+      ! Go through all 4 corner nodes
+      Side%CurvedNode(QuadMapInv(0,0))%np%tmp = 0
+      Side%CurvedNode(QuadMapInv(0,N))%np%tmp = 0
+      Side%CurvedNode(QuadMapInv(N,0))%np%tmp = 0
+      Side%CurvedNode(QuadMapInv(N,N))%np%tmp = 0
+    ELSE IF(Side%BC%BCType.EQ.1) THEN
+      ! Go through all 4 corner nodes
+      Side%CurvedNode(QuadMapInv(0,0))%np%tmp = 0
+      Side%CurvedNode(QuadMapInv(0,N))%np%tmp = 0
+      Side%CurvedNode(QuadMapInv(N,0))%np%tmp = 0
+      Side%CurvedNode(QuadMapInv(N,N))%np%tmp = 0
+    ELSE
+      DO i=1,Side%nCurvedNodes
+        Side%CurvedNode(i)%np%tmp = 0
+      END DO
+    END IF
+    Side=>Side%nextElemSide
+  END DO
+  Elem=>Elem%nextElem
+END DO
+
+nBP = 0
+Elem=>FirstElem
+DO WHILE(ASSOCIATED(Elem))
+  Side=>Elem%firstSide
+  DO WHILE(ASSOCIATED(Side))
+    IF(.NOT.ASSOCIATED(Side%BC)) THEN
+      ! Not a BC, include the corner nodes
+      IF (.NOT.(Side%CurvedNode(QuadMapInv(0,0))%np%tmp.EQ.1)) THEN
+        ! Check if node is inside the box
+        x = Side%CurvedNode(QuadMapInv(0,0))%np%x
+        IF ((x(1).LT.xMin).OR.(x(1).GT.xMax).OR.(x(2).LT.yMin).OR.(x(2).GT.yMax)) EXIT
+        ! Increase count of boundary points if inside the box
+        nBP = nBP + 1
+        ! Mark this side node as done
+        Side%CurvedNode(QuadMapInv(0,0))%np%tmp = 1
+      END IF
+      IF (.NOT.(Side%CurvedNode(QuadMapInv(0,N))%np%tmp.EQ.1)) THEN
+        x = Side%CurvedNode(QuadMapInv(0,0))%np%x
+        IF ((x(1).LT.xMin).OR.(x(1).GT.xMax).OR.(x(2).LT.yMin).OR.(x(2).GT.yMax)) EXIT
+        nBP = nBP + 1
+        Side%CurvedNode(QuadMapInv(0,N))%np%tmp = 1
+      END IF
+      IF (.NOT.(Side%CurvedNode(QuadMapInv(N,0))%np%tmp.EQ.1)) THEN
+        x = Side%CurvedNode(QuadMapInv(0,0))%np%x
+        IF ((x(1).LT.xMin).OR.(x(1).GT.xMax).OR.(x(2).LT.yMin).OR.(x(2).GT.yMax)) EXIT
+        nBP = nBP + 1
+        Side%CurvedNode(QuadMapInv(N,0))%np%tmp = 1
+      END IF
+      IF (.NOT.(Side%CurvedNode(QuadMapInv(N,N))%np%tmp.EQ.1)) THEN
+        x = Side%CurvedNode(QuadMapInv(0,0))%np%x
+        IF ((x(1).LT.xMin).OR.(x(1).GT.xMax).OR.(x(2).LT.yMin).OR.(x(2).GT.yMax)) EXIT
+        nBP = nBP + 1
+        Side%CurvedNode(QuadMapInv(N,N))%np%tmp = 1
+      END IF
+    ELSE IF(Side%BC%BCType.EQ.1) THEN
+      ! Periodic BC, threat like a normal (non-BC) side
```

Comment:
What happens here if the box only covers on periodic side but not the other? DG does not care but FLEXI has multiple branches operating in physical space.

Reply:
RESOLVED: Call ABORT

Reply:
The RBF Curving only works in an xy-plane and thus mainly for extruded 2D meshes. Here, periodic boundaries in z-direction are natural and should work flawlessly. I think the consistency of periodic sides should be ensured by the users for their specific cases. HOPR should also fail, since it should not be able to connect the curved/uncurved partner sides.